src/apply-names.cc

```diff
@@ -43,35 +42,33 @@ struct Context {
   /* mapping from param index to its name, if any, for the current func */
   std::vector<std::string> param_index_to_name;
   std::vector<std::string> local_index_to_name;
-  LabelPtrVector labels;
+  std::vector<Label*> labels;
 };
 
 Context::Context() {
   WABT_ZERO_MEMORY(visitor);
-  WABT_ZERO_MEMORY(labels);
 }
 
 void push_label(Context* ctx, Label* label) {
-  append_label_ptr_value(&ctx->labels, &label);
+  ctx->labels.push_back(label);
 }
 
 void pop_label(Context* ctx) {
-  assert(ctx->labels.size > 0);
-  ctx->labels.size--;
+  ctx->labels.pop_back();
 }
 
 Label* find_label_by_var(Context* ctx, Var* var) {
   if (var->type == VarType::Name) {
-    for (int i = ctx->labels.size - 1; i >= 0; --i) {
-      Label* label = ctx->labels.data[i];
+    for (int i = ctx->labels.size() - 1; i >= 0; --i) {
+      Label* label = ctx->labels[i];
       if (string_slices_are_equal(label, &var->name))
         return label;
     }
     return nullptr;
   } else {
-    if (var->index < 0 || static_cast<size_t>(var->index) >= ctx->labels.size)
+    if (var->index < 0 || static_cast<size_t>(var->index) >= ctx->labels.size())
```

Comment:
Hm, true. Though I kind of like that it's easier to grep for static_cast. In reality, it might be best to change var->index to a size_t since this ends up happening all over.